SimPEG/Regularization.py

```diff
@@ -318,11 +318,20 @@ def __init__(self, mesh, mapping=None, indActive=None, **kwargs):
         if indActive is not None and mapping is None:
             mapping = Maps.IdentityMap(nP=indActive.nonzero()[0].size)
 
-        self.regmesh = RegularizationMesh(mesh,indActive)
-        self.mapping = mapping or self.mapPair(mesh)
-        self.mapping._assertMatchesPair(self.mapPair)
+        if mesh is None and nP is None:
+            raise Exception('either Mesh or number of parameters must be '
+                            'provided to the BaseRegularization')
+
+        self.regmesh = RegularizationMesh(mesh, indActive)
```

Comment:
we have built towards it, but at this point, it shouldn't work (the only case it would work is if you use smallness only)